packages/parser-sample/src/performance-demo.ts

```
@@ -0,0 +1,261 @@
+  }
+}
+
+console.log(`\n${"=".repeat(60)}\n`);
+
+// Memory usage comparison
+console.log("💾 Memory Usage Analysis...");
+
+function measureMemory(operation: () => void, iterations: number = 100): number {
+  if (global.gc) global.gc();
+  
+  const before = process.memoryUsage().heapUsed;
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)